### PR TITLE
Redesign the front page and base layout of the site.

### PR DESCRIPTION
**Pull request by n-rook**
This was created by taking #306 and converting the changes made in it to change, rather than replacing, our existing SASS stylesheets. This will limit the amount of existing HTML that breaks accidentally, and will simplify further work.

I also left out some of the changes in #306. This is mostly to simplify code review: A lot of the changes seemed unrelated, and it was really hard for me to evaluate and judge them all at the same time. (It's largely not because I thought the changes were bad ideas.) This is why I submitted a new PR rather than just committing to #306, since I figure a lot of the changes not in this PR are still things we'll want to do eventually.

### CODE REVIEW

**Comment by n-rook** [comment]
image
image
image

File: project/thscoreboard/replays/testing/test_utilities.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    necessary to get consistent results if we want to test functions that
+    return translated strings.
+
+    Note that there is probably something wrong with the way we're
+    running translations in CI, and there's probably a better way to do this.
+    See https://github.com/n-rook/thscoreboard/issues/58
+    """
+    # https://github.com/n-rook/thscoreboard/issues/58
+    # gettext doesn't work right on the GitHub bot, so we intentionally
+    # override it with a language for which we don't have translations.
```
Comment (jgdhs27) [inline project/thscoreboard/replays/testing/test_utilities.py:18]: remove the original comment

File: project/thscoreboard/shared_content/static/favicon.png
Comment (jgdhs27) [inline project/thscoreboard/shared_content/static/favicon.png:1]: do we need it in png form?

File: project/thscoreboard/shared_content/static/style.sass [hunk 2]
```diff
@@ -3,52 +3,126 @@
...
-    background-color: var(--topbar-color)
-    min-height: 50px
-    display: flex
-    align-items: center
-    margin: 0 0 5px
+body::before
+    content: ''
+    position: absolute
+    inset: 0
+    background-image: url(favicon.png)
```
Comment (jgdhs27) [inline project/thscoreboard/shared_content/static/style.sass:15]: this doesn't look great on dark mode

File: project/thscoreboard/thscoreboard/templates/base.html [hunk 3]
```diff
@@ -2,62 +2,80 @@
...
+                    <a href="/users/profile">Profile</a>
+                </div>
+                {% if user.is_staff or user.is_superuser %}
+                <a href="/staff">Staff tools</a>
+                {% endif %}
+                {% else %}
+                <a href="/users/login">Log in</a>
+                <a href="/users/register">Sign Up</a>
+                {% endif %}
+                <a href="#" onclick="toggleTheme()">🌙</a>
```
Comment (jgdhs27) [inline project/thscoreboard/thscoreboard/templates/base.html:43]: could we make this not be underlined?

**Review by jgdhs27 — CHANGES_REQUESTED** [review]
* A couple of things seems broken in dark mode. The background image looks ugly:
image
And some highlights are missing (compare with light mode):
image

* The headers look uncomfortably close to the left side of the box, and not the top
image

* Please change "(Edit)" in replay_details.html to "(Edit comment)", it looks a bit strange especially when the replay doesnt have a comment.

**Comment by n-rook** [comment]
image
image

**Comment by n-rook** [comment]
Done. I fixed (or at least tried to fix) everything mentioned above, except for the (edit) thing which is out of scope of this PR. (I agree it should be changed, but it's not really related to this change.)

**Comment by jgdhs27** [comment]
image
the background is still a bit broken on dark mode if you scroll down

**Comment by jgdhs27** [comment]
One more thing i just noticed: some fields in the login and sign up page are unstyled
image
image

**Comment by n-rook** [comment]
Both of those should be fixed now

**Review by jgdhs27 — COMMENTED** [review]
Yay! Thanks for all the work that went into this, it looks amazing!

**Review by jgdhs27 — APPROVED** [review]
Yay! Thanks for all the work that went into this, it looks amazing!

**Pull request merged**